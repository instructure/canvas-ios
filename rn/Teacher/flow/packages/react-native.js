//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
