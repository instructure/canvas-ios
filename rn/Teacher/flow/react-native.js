//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
