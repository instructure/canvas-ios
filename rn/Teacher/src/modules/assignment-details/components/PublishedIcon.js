//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/**
 * @flow
 */

import React from 'react'
import { View, Image } from 'react-native'
import Images from '../../../images'
import i18n from 'format-message'
import { Text } from '../../../common/text'
import { createStyleSheet } from '../../../common/stylesheet'

export default class PublishedIcon extends React.Component<Object> {
  render () {
    let { published, style, iconSize } = this.props

    const publishedIcon = published ? Images.published : Images.unpublished

    let iconStyle = published ? internalStyle.publishedIcon : internalStyle.unpublishedIcon
    let statusID = published ? 'published' : 'unpublished'
    let customStyle = {}
    if (iconSize) {
      customStyle = { width: iconSize, height: iconSize }
    }

    let unpublishedText = i18n('Unpublished')

    let publishedText = i18n('Published')

    return (
      <View style={[ style, internalStyle.container ]}>
        <Image
          source={publishedIcon}
          style={[iconStyle, customStyle]}
          testID={`assignment-details.published-icon.${statusID}-status-img`}
        />
        <Text
          style={[internalStyle.text, published ? internalStyle.publishedTextStyle : internalStyle.unPublishedTextStyle]}
          testID='assignment-details.published-icon.publish-status-lbl'
        >
          {(published) ? publishedText : unpublishedText}
        </Text>
      </View>
    )
  }
}

const iconSize = 21

const internalStyle = createStyleSheet(colors => ({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    marginLeft: 6,
    fontWeight: '500',
    color: colors.textDark,
  },
  publishedTextStyle: { color: colors.textSuccess },
  unPublishedTextStyle: { color: colors.textDark },
  publishedIcon: {
    height: iconSize,
    width: iconSize,
    tintColor: colors.textSuccess,
  },
  unpublishedIcon: {
    height: iconSize,
    width: iconSize,
    tintColor: colors.textDark,
  },
}))
