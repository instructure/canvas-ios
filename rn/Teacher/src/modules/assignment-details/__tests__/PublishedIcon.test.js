//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/**
 * @flow
 */

import 'react-native'
import React from 'react'
import PublishedIcon from '../components/PublishedIcon'
import renderer from 'react-test-renderer'

test('render published', () => {
  let tree = renderer.create(
    <PublishedIcon published={true} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render unpublished', () => {
  let tree = renderer.create(
    <PublishedIcon published={false} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render larger icon size', () => {
  let tree = renderer.create(
    <PublishedIcon published={true} iconSize={75} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
