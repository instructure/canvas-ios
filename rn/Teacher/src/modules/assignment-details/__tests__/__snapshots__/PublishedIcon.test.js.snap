// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render larger icon size 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/publish.png",
      }
    }
    style={
      Array [
        Object {
          "height": 21,
          "tintColor": "#127A1B",
          "width": 21,
        },
        Object {
          "height": 75,
          "width": 75,
        },
      ]
    }
    testID="assignment-details.published-icon.published-status-img"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#2D3B45",
          "fontFamily": "Lato-Regular",
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#556572",
            "fontWeight": "500",
            "marginLeft": 6,
          },
          Object {
            "color": "#127A1B",
          },
        ],
      ]
    }
    testID="assignment-details.published-icon.publish-status-lbl"
  >
    Published
  </Text>
</View>
`;

exports[`render published 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/publish.png",
      }
    }
    style={
      Array [
        Object {
          "height": 21,
          "tintColor": "#127A1B",
          "width": 21,
        },
        Object {},
      ]
    }
    testID="assignment-details.published-icon.published-status-img"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#2D3B45",
          "fontFamily": "Lato-Regular",
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#556572",
            "fontWeight": "500",
            "marginLeft": 6,
          },
          Object {
            "color": "#127A1B",
          },
        ],
      ]
    }
    testID="assignment-details.published-icon.publish-status-lbl"
  >
    Published
  </Text>
</View>
`;

exports[`render unpublished 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/unpublish.png",
      }
    }
    style={
      Array [
        Object {
          "height": 21,
          "tintColor": "#556572",
          "width": 21,
        },
        Object {},
      ]
    }
    testID="assignment-details.published-icon.unpublished-status-img"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#2D3B45",
          "fontFamily": "Lato-Regular",
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#556572",
            "fontWeight": "500",
            "marginLeft": 6,
          },
          Object {
            "color": "#556572",
          },
        ],
      ]
    }
    testID="assignment-details.published-icon.publish-status-lbl"
  >
    Unpublished
  </Text>
</View>
`;
